carsus/io/cmfgen/base.py

```diff
@@ -464,3 +467,96 @@ def to_hdf(self, key='/photoionization_cross_sections'):
+        Class for extracting levels and lines from CMFGEN.
+        
+        Mimics the GFALLReader class.
+
+        Attributes
+        ----------
+        levels : DataFrame
+        lines : DataFrame
+
+    """
+
+    def __init__(self, data, priority=10):
+        """
+        Parameters
+        ----------
+        data : dict
+            Dictionary containing one dictionary per specie with 
+            keys `levels` and `lines`.
+
+        priority: int, optional
+            Priority of the current data source, by default 10.
+        """
+        self.priority = priority
+        self._get_levels_lines(data)
+    
+    def _get_levels_lines(self, data):
+        """ Generates `levels` and `lines` DataFrames.
+
+        Parameters
+        ----------
+        data : dict
+            Dictionary containing one dictionary per specie with 
+            keys `levels` and `lines`.
+        """
+        lvl_list = []
+        lns_list = []
+        for ion, parser in data.items():
+
+            atomic_number = parse_selected_species(ion)[0][0]
+            ion_charge = parse_selected_species(ion)[0][1]
+            
+            lvl = parser['levels'].base
+            # some ID's have negative values (theoretical?)
+            lvl.loc[ lvl['ID'] < 0, 'method'] = 'theor'
+            lvl.loc[ lvl['ID'] > 0, 'method'] = 'meas'
+            lvl['ID'] = lvl['ID'].apply(np.abs)
+            lvl_id = lvl.set_index('ID')
+            lvl['atomic_number'] = atomic_number
+            lvl['ion_charge'] =  ion_charge
+            lvl_list.append(lvl)
+
+            lns = parser['lines'].base
+            lns = lns.set_index(['i', 'j'])
+            lns['energy_lower'] = lvl_id['E(cm^-1)'].reindex(lns.index, level=0).values
+            lns['energy_upper'] = lvl_id['E(cm^-1)'].reindex(lns.index, level=1).values
+            lns['g_lower'] = lvl_id['g'].reindex(lns.index, level=0).values
+            lns['g_upper'] = lvl_id['g'].reindex(lns.index, level=1).values
+            lns['j_lower'] = (lns['g_lower'] -1)/2
+            lns['j_upper'] = (lns['g_upper'] -1)/2
+            lns['atomic_number'] = atomic_number
+            lns['ion_charge'] = ion_charge
+            lns = lns.reset_index()
+            lns_list.append(lns)
+
+        levels = pd.concat(lvl_list)
+        levels['priority'] = self.priority
+        levels = levels.reset_index(drop=False)
+        levels = levels.rename(columns={'Configuration': 'label', 
+                                        'E(cm^-1)': 'energy', 
+                                        'index': 'level_index'})
+        levels['j'] = (levels['g'] -1)/2
+        levels = levels.set_index(['atomic_number', 'ion_charge', 'level_index'])
+        levels = levels[['energy', 'j', 'label', 'method', 'priority']]
+        
+        lines = pd.concat(lns_list)
+        lines = lines.rename(columns={'Lam(A)': 'wavelength'})
+        lines['wavelength'] = lines['wavelength'].apply(lambda x: (x*u.AA).to('nm').value)
```

Comment:
after merging we should run `black`

Reply:
why do we do `nm` instead of `angstrom`?

Reply:
if we decide to go with nm - we should do that
```suggestion
        lines['wavelength'] = lines['wavelength'] * u.AA.to(u.nm)
```

Reply:
- Yes, please, we need `black`.
- It's done because GFALL and Chianti readers returns `nm` in their DataFrames, it's just for consistency. Units are managed with the highest possible care and consistency, but there are still some things to fix.

Reply:
We should use @wkerzendorf's suggestion for the unit conversion.

Reply:
Documented at https://github.com/tardis-sn/carsus/issues/215. Before this I want to merge the tests for the `output` module in the upcoming PR.

Reply:
sounds good!